cmd/ethrex/cli.rs

```
@@ -350,13 +352,25 @@ pub async fn import_blocks(path: &str, data_dir: &str, network: &str, evm: EvmEn
             "Adding block {} with hash {:#x}.",
             block.header.number, hash
         );
-
-        if let Err(error) = blockchain.add_block(block).await {
-            warn!(
-                "Failed to add block {} with hash {:#x}: {}.",
-                block.header.number, hash, error
-            );
-            return;
+        // Check if the block is already in the blockchain, if it is do nothing, if not add it
+        match store.get_block_number(hash).await {
+            Ok(Some(_)) => {
+                info!("Block {} is already in the blockchain", block.hash());
+                continue;
+            }
+            Ok(None) => {
+                if let Err(error) = blockchain.add_block(block).await {
+                    warn!(
+                        "Failed to add block {} with hash {:#x}: {}.",
+                        block.header.number, hash, error
+                    );
+                    return;
+                }
+            }
+            Err(_) => {
```

maybe `import_blocks` should return a Result and here you should return an Error

---

Done! Now `import_blocks` returns `Result<(),ChainErrors>` which is the same that `add_block` returns.
Decided to still print the warning message inside `import_blocks` if add_block produced an error, and then return that same error.